### ISSUE TITLE
Crash when autocomplete enum on int key dictionary

### ISSUE THREAD

**Issue by orblazer** [godotengine/godot#72936]
### Godot version

v4.0.beta17.official [c40020513]

### System information

Linux 5.15.91-1-MANJARO

### Issue description

When try to autocomplete enum on int key in dictionary, Godot crash.

### Steps to reproduce

1. Create an `gdscript`
2. put this code, and follow the comment

    ```py
    enum Power { NORMAL, CHARGED, CRIT }

    const frame_to_power = {
        1: Power. # [ctrl+space] here
    }
    ```

### Minimal reproduction project

N/A

**Comment by vonagam** [comment]
I assume this one is fixed by #72557. Can you try release candidate?

**Comment by anvilfolk** [comment]
Confirmed fixed in RC1, recommend closing, @YuriSizov 

![image](https://user-images.githubusercontent.com/1133892/217836912-3d0e878f-517f-4726-a9ee-2aa36e4cbb93.png)
